utils/src/lib.rs

```diff
@@ -45,103 +274,107 @@ impl MemoryMetrics {
                     virtual_mem: Some(new_memory_stats.virtual_mem),
                     physical_mem_diff,
                     virtual_mem_diff,
+                    allocated,
+                    deallocated,
+                    alloc_calls,
+                    peak,
+                    in_use,
+                    elapsed,
                 }
             }
             (None, Some(new_memory_stats)) => Self {
                 physical_mem: Some(new_memory_stats.physical_mem),
                 virtual_mem: Some(new_memory_stats.virtual_mem),
                 physical_mem_diff: None,
                 virtual_mem_diff: None,
+                allocated,
+                deallocated,
+                alloc_calls,
+                peak,
+                in_use,
+                elapsed,
             },
             (None, None) | (Some(_), None) => Self {
                 physical_mem: None,
                 virtual_mem: None,
                 physical_mem_diff: None,
                 virtual_mem_diff: None,
+                allocated,
+                deallocated,
+                alloc_calls,
+                peak,
+                in_use,
+                elapsed,
             },
         }
     }
 }
 
-pub struct Metrics {
-    pub elapsed: Duration,
-    pub memory: MemoryMetrics,
-}
+impl Display for MetricsSpan {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "elapsed={:?}", self.elapsed)?;
 
-#[must_use]
-pub struct MeasureStage<'a> {
-    close: Cow<'a, str>,
-    start: Instant,
-    memory_stats: Option<MemoryStats>,
-}
+        write!(f, " peak=")?;
+        if let Some(peak) = self.peak {
+            format_bytes_usize(f, peak)?;
+        };
 
-pub struct Guard<'a> {
-    measure: MeasureStage<'a>,
-}
+        write!(f, " in_use=")?;
+        if let Some(in_use) = self.in_use {
+            format_bytes_usize(f, in_use)?;
+        };
 
-impl<'a> Drop for Guard<'a> {
-    fn drop(&mut self) {
-        let _ = self.measure.metrics();
-    }
-}
+        if f.alternate() {
```

Comment:
By default only the elapsed time, peak, and current memory use are printed. This makes for a cleaner logging.